### DISCUSSION TITLE
Set ~ (default home dir) in wsl

### DISCUSSION THREAD

**Discussion by zallaevan** [GitSquared/edex-ui#976] [Questions] [answered]
I wanted to set the cwd to ~ dir, so it starts in my home dir. I use the wsl.exe terminal, and I have tried, but it just crashes. If I set cwd to windows dir (C:/Temp) it works, but if I use \\wsl$\Ubuntu-20.04\root (home dir as in explorer.exe) it doesn't work.

Thanks in advance.

**Comment by GitSquared** [comment]
Maybe [this](https://github.com/GitSquared/edex-ui/issues/778#issuecomment-724000098) can help?

**Reply by zallaevan** [reply]
Nope, it doesn't work. As I mentioned earlier, I'd set it to `\\wsl$\Ubuntu-20.04\root` and to `\wsl$\Ubuntu-20.04\root`. However, it doesn't start and doesn't show the GUI.

**Reply by GitSquared** [reply]
Could you try running edex's binary .exe from a terminal/powershell window and see what kind of logs you get there?

**Reply by zallaevan** [reply]
When I try with `\wsl$...` it shows this:

Ôû║  start     Starting eDEX-UI v2.2.5
i  info      With Node 12.16.3 and Electron 10.2.0
i  info      Renderer is Chrome 85.0.4183.121
Ôû║  Startup   Initialized timer...
i  info      Base config dir is C:\Users\zallaevan\AppData\Roaming\eDEX-UI
[ ]  pending   Mirroring internal assets...
[ ]  pending   Loading settings file...
(node:4796) UnhandledPromiseRejectionWarning: SyntaxError: C:\Users\zallaevan\AppData\Roaming\eDEX-UI\settings.json: Unexpected token w in JSON at position 60
    at parse (<anonymous>)
    at Object.Module._extensions..json (internal/modules/cjs/loader.js:1180:22)
    at Module.load (internal/modules/cjs/loader.js:981:32)
    at Module._load (internal/modules/cjs/loader.js:881:14)
    at Function.Module._load (electron/js2c/asar.js:769:28)
    at Module.require (internal/modules/cjs/loader.js:1023:19)
    at require (internal/modules/cjs/helpers.js:77:18)
    at App.<anonymous> (C:\Program Files\eDEX-UI\resources\app.asar\_boot.js:219:20)
    at App.emit (events.js:310:20)
(node:4796) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag --unhandled-rejections=strict (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:4796) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

However, when I try `\\wsl$...`, this happens:

Ôû║  start     Starting eDEX-UI v2.2.5
i  info      With Node 12.16.3 and Electron 10.2.0
i  info      Renderer is Chrome 85.0.4183.121
Ôû║  Startup   Initialized timer...
i  info      Base config dir is C:\Users\zallaevan\AppData\Roaming\eDEX-UI
[ ]  pending   Mirroring internal assets...
[ ]  pending   Loading settings file...
(node:17120) UnhandledPromiseRejectionWarning: SyntaxError: C:\Users\zallaevan\AppData\Roaming\eDEX-UI\settings.json: Unexpected token U in JSON at position 66
    at parse (<anonymous>)
    at Object.Module._extensions..json (internal/modules/cjs/loader.js:1180:22)
    at Module.load (internal/modules/cjs/loader.js:981:32)
    at Module._load (internal/modules/cjs/loader.js:881:14)
    at Function.Module._load (electron/js2c/asar.js:769:28)
    at Module.require (internal/modules/cjs/loader.js:1023:19)
    at require (internal/modules/cjs/helpers.js:77:18)
    at App.<anonymous> (C:\Program Files\eDEX-UI\resources\app.asar\_boot.js:219:20)
    at App.emit (events.js:310:20)
(node:17120) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:17120) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

**Reply by zallaevan** [reply]
Sorry for this long response. Even though I change `Ubuntu` to `ubuntu`, it still shows me an error.

**Reply by Animus-Surge** [reply]
Are you setting the default dir path to something like this: `//wsl$/Ubuntu/home/<username>`?

EDIT: because I tried with the same thing, only replacing `/home/<username>` with `/root` and it didn't work. It only dropped me to `/` and not `~`

**Reply by GitSquared** [reply]
> Unexpected token w in JSON at position 60

\\**w**

> Unexpected token U in JSON at position 66

\\\wsl$\\**u**buntu

I think your problem here might be that JSON needs you to escape all backslashes by doubling them, like so: `\\wsl$\\Ubuntu-20.04\\root`

**Reply by zallaevan** [reply]
Okay, I figured out how to solve it, thanks to @Animus-Surge. I changed cwd to `//wsl$/Ubuntu-20.04/root` instead of `\\wsl$\Ubuntu-20.04\root` and it worked like a charm. Also, I tried your solution, @GitSquared, `\\wsl$\\Ubuntu-20.04\\root`, but it throws a `not found` error. Thank you for your time.

**Reply by GitSquared** [reply]
Cheers, glad we could figure this out!

**Comment by zallaevan** [comment, marked as answer]
> Are you setting the default dir path to something like this: `//wsl$/Ubuntu/home/<username>`?
> 
> EDIT: because I tried with the same thing, only replacing `/home/<username>` with `/root` and it didn't work. It only dropped me to `/` and not `~`

Comment from @Animus-Surge was the solution. Instead of choosing `\\wsl$\<path>`, it worked with **`//wsl$/<path>`**.
